test/e2e/e2e_test.go

```diff
@@ -105,6 +105,35 @@ var _ = ginkgo.Describe("JobSet", func() {
 			util.JobSetCompleted(ctx, k8sClient, js, timeout)
 		})
 	})
+	ginkgo.When("ttl seconds after finished is set", func() {
+		ginkgo.It("should clean up the completed jobset after configured ttl seconds expire", func() {
+			ctx := context.Background()
+
+			// Create JobSet.
+			testFinalizer := "fake.example.com/blockDeletion"
+			ginkgo.By("creating jobset with ttl seconds after finished")
+			js := sleepTestJobSet(ns).Finalizers([]string{testFinalizer}).TTLSecondsAfterFinished(5).Obj()
+
+			// Verify jobset created successfully.
+			ginkgo.By("checking that jobset creation succeeds")
+			gomega.Expect(k8sClient.Create(ctx, js)).Should(gomega.Succeed())
+
+			// Check jobset status if specified.
+			ginkgo.By("checking jobset condition")
+			util.JobSetCompleted(ctx, k8sClient, js, timeout)
+
+			// We get the latest version of the jobset before removing the finalizer
+			var fresh jobset.JobSet
+			gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &fresh)).Should(gomega.Succeed())
```

Comment:
move this get to inside removeJobSetFinalizer within Eventually (see comment there)